internal/controllers/admission_test.go

```diff
@@ -19,159 +19,178 @@ func operator(spec operatorsv1alpha1.OperatorSpec) *operatorsv1alpha1.Operator {
 	}
 }
 
-var _ = Describe("Operator Spec Validations", func() {
-	var (
-		ctx    context.Context
-		cancel context.CancelFunc
-	)
-	BeforeEach(func() {
-		ctx, cancel = context.WithCancel(context.Background())
-	})
-	AfterEach(func() {
-		cancel()
-	})
-	It("should fail if the spec is empty", func() {
-		err := cl.Create(ctx, operator(operatorsv1alpha1.OperatorSpec{}))
-		Expect(err).To(HaveOccurred())
-		Expect(err.Error()).To(ContainSubstring("spec.packageName in body should match '^[a-z0-9]+(-[a-z0-9]+)*$'"))
-	})
-	It("should fail if package name length is greater than 48 characters", func() {
-		err := cl.Create(ctx, operator(operatorsv1alpha1.OperatorSpec{
-			PackageName: "this-is-a-really-long-package-name-that-is-greater-than-48-characters",
-		}))
-		Expect(err).To(HaveOccurred())
-		Expect(err.Error()).To(ContainSubstring("Too long: may not be longer than 48"))
-	})
-	It("should fail if version is valid semver but length is greater than 64 characters", func() {
+func TestOperatorSpecIsEmpty(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := cl.Create(ctx, operator(operatorsv1alpha1.OperatorSpec{}))
```

Comment:
Where does `cl` come from? is there a factoring that allows this client to be constructed at runtime for the test itself?

Reply:
`cl` comes from the test_suite.go file, so it's global for the test program. Wouldn't creating `cl` for each test cause a lengthening of the test duration (all those extra `client.New()`s)? Again, doing a direct translation to reduce the risk of deleing/breaking tests. Optimizations can be done later.

Reply:
`client.New()` should be an allocation or two - and cost little.